### ISSUE TITLE
make batch norm in place by default

### ISSUE THREAD

**Issue by tonyyang-svail** [PaddlePaddle/Paddle#9412]
Batchnorm has been treated as an in-place operator in [official ResNet](https://github.com/KaimingHe/deep-residual-networks/blob/master/prototxt/ResNet-50-deploy.prototxt#L75-L76). Better visualization [here](http://ethereon.github.io/netscope/#/gist/3df1f508f14651cf127e9fbf009453ca).

For vision net, this would make a significant improvement in memory usage.
